### ISSUE TITLE
Make "Solo Mode" setting work in AR

### ISSUE THREAD

**Issue by electrofloat** [jixxed/ed-odyssey-materials-helper#104]
I'm so much annoyed that the "Solo Mode" setting does not work in AR that I tried to fix it myself.
Of course not knowing java/gradle, it failed miserably.

How do you compile this?

I tried `gradlew createPortableRelease` but it failed with:
```
> Task :bootstrap:createDelegatingModules FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':bootstrap:createDelegatingModules'.
> java/lang/module/ModuleFinder
```

I don't have a single clue of what that means.

Anyways something like this should - I believe - fix the issue I have:
```
diff --git a/application/src/main/java/nl/jixxed/eliteodysseymaterials/constants/OdysseyBlueprintConstants.java b/application/src/main/java/nl/jixxed/eliteodysseymaterials/constants/OdysseyBlueprintConstants.java
index 4d47765..0219ba6 100644
--- a/application/src/main/java/nl/jixxed/eliteodysseymaterials/constants/OdysseyBlueprintConstants.java
+++ b/application/src/main/java/nl/jixxed/eliteodysseymaterials/constants/OdysseyBlueprintConstants.java
@@ -116,7 +116,9 @@ public abstract class OdysseyBlueprintConstants {
     }

     public static boolean isEngineeringOrBlueprintIngredientWithOverride(final OdysseyMaterial odysseyMaterial) {
-        return isEngineeringIngredient(odysseyMaterial) || isBlueprintIngredientWithOverride(odysseyMaterial);
+        final boolean soloMode = PreferencesService.getPreference(PreferenceConstants.SOLO_MODE, Boolean.FALSE);
+        final boolean isIngredient = soloMode ? isEngineeringIngredientAndNotCompleted(odysseyMaterial) : isEngineeringIngredient(odysseyMaterial);
+        return isIngredient || isBlueprintIngredientWithOverride(odysseyMaterial);
     }

     public static boolean isEngineeringIngredient(final OdysseyMaterial odysseyMaterial) {
```

Can you confirm/tell me how to compile the project?

Thanks.

**Comment by jixxed** [comment]
I don't know what that error means either. In any case you only need to run the project inside the application directory. the bootstrap project is for the autoupdater. in the application project you want to run the "run" target. that launches in second, while createPortableRelease takes minutes to compile.
![image](https://user-images.githubusercontent.com/14129491/204094099-b4a940fd-5fba-4ea9-bc93-a30ce0e59e18.png)
Running from an IDE like IntelliJ IDEA may make things a bit easier.

In any case I will have a look into this issue myself too.

**Comment by electrofloat** [comment]
Thanks!

running `gradlew run` in the application folder worked.

Also it seems that the modification worked. For the first time the "Smear Campaign Plans" was shown as irrelevant at last :)

I'm not sure if this is the best way to have it check for solo mode, since I found a couple of other places where you check for that, for example in OdysseyMaterialShow.java you alread have a:
```
    private static boolean getIrrelevantFilter(final OdysseyMaterial odysseyMaterial) {
        return APPLICATION_STATE.getSoloMode() ? OdysseyBlueprintConstants.isNotRelevantWithOverrideAndNotRequiredEngineeringIngredient(odysseyMaterial) : OdysseyBlueprintConstants.isNotRelevantAndNotEngineeringIngredient(odysseyMaterial);
    }
```

but for now it does what I wanted it to do, so until you have time to do it your way, this is good enough for me.

**Comment by jixxed** [comment]
just released the fix in 2.1
was going to ask you to verify it, since I am not near the bubble. good to know it fixes the issue
OdysseyMaterialCard:createMaterialImage does the same thing you want, so that is a good example
APPLICATION_STATE.getSoloMode() is the preferred way, but it does the same thing.
